From 879f5d6f9f10ccb0264cdedfe95eac15fb3ed94c Mon Sep 17 00:00:00 2001
From: Ola Olsson <olax.olsson@intel.com>
Date: Mon, 7 Nov 2011 10:42:26 -0800
Subject: [PORT FROM MAIN] libcorkscrew: Fix hang during backtrace

BZ: 13909

Skip lookups of addresses if the symbol table is not found during
tombstone generation

Change-Id: Ib3f47a1a5b3c2f4c6b762d5a592ce0e76a45aabc
Orig-Change-Id: Ic06cc9c6cd0ad854aa62bbeba82b3688b4d39e5e
Category: device enablement
Domain: AOSP-Bionic
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Bruce Beare <bruce.j.beare@intel.com>
Reviewed-on: http://android.intel.com:8080/78856
---
 libcorkscrew/symbol_table.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libcorkscrew/symbol_table.c b/libcorkscrew/symbol_table.c
index 982ccc8..c9b7e02 100644
--- a/libcorkscrew/symbol_table.c
+++ b/libcorkscrew/symbol_table.c
@@ -98,7 +98,7 @@ symbol_table_t* load_symbol_table(const char *filename) {
             dynsym_idx = i;
         }
     }
-    if (dynsym_idx == -1 && sym_idx == -1) {
+    if (dynsym_idx == -1 || sym_idx == -1) {
         goto out_unmap;
     }
 
-- 
1.7.4.1

